src/utils.js

```diff
@@ -39,43 +38,37 @@ var getCommentFormatter = function (file) {
 
   switch (extension) {
     case 'css':
-      debug(logCb('css commentFormatter'));
+      debug(function() { return 'css commentFormatter';});
       commentFormatter = function(url) {
         return newline + "/*# sourceMappingURL=" + url + " */" + newline;
       };
       break;
     case 'js':
-      debug(logCb('js commentFormatter'));
+      debug(function() { return 'js commentFormatter'; });
       commentFormatter = function(url) {
         return newline + "//# sourceMappingURL=" + url + newline;
       };
       break;
     default:
-      debug(logCb('unknown commentFormatter'));
+      debug(function() { return 'unknown commentFormatter'; });
   }
 
   return commentFormatter;
 };
 
 var getInlinePreExisting = function(fileContent){
   if(sourceMapUrlRegEx().test(fileContent)){
-    debug(logCb('has preExisting'));
+    debug(function() { return 'has preExisting'; });
```

Comment:
That's a good question, I am sure that if they were not closures and defined / named functions then it would be more optimal.

I guess I should come up with some specs in `debug-fabulous` to compare. Anyway a good rule would be if the string concatenation is large use a closure. I am just used to doing this everywhere as we did this at work heavily.